client/network/src/service.rs

```diff
@@ -1164,7 +1167,34 @@ struct Metrics {
 	requests_out_started_total: CounterVec<U64>,
 }
 
+/// The source for bandwidth metrics.
+#[derive(Clone)]
+struct BandwidthSource(Arc<transport::BandwidthSinks>);
+
+impl MetricSource for BandwidthSource {
+	type N = u64;
+
+	fn collect(&self, mut set: impl FnMut(&[&str], Self::N)) {
+		set(&[&"in"], self.0.total_inbound());
+		set(&[&"out"], self.0.total_outbound());
+	}
+}
+
 impl Metrics {
+	fn register_bandwidth(registry: &Registry, sinks: Arc<transport::BandwidthSinks>)
```

Comment:
I am a bit torn whether this should be part of `Metrics` as in the end `Metrics` doesn't even keep a reference to the `BandwidthSource`. What do you think of doing the registration in the `// Initialize the metrics.` block above right away? Again, not sure that is easier to read.

Reply:
I still think it preferable to not bury the definition of a metric in other control flow. Is https://github.com/paritytech/substrate/pull/6895/commits/22be2f76fd5d07ae390b9d5ab92e86f0c8e251e2 acceptable to you? (I.e. `Metrics::register_bandwidth` -> `BandwidthCounters::register`).